File: Sort/quick.c
//快速排序

#include <stdio.h>

int a[101], n;

void quick(int left, int right)
{
    int i, j,flag, temp;
    if (left > right)
        return;
    flag = a[left];
    i = left;
    j = right;

    while (i != j)
    {
        while (a[j] >= flag && j > i)
            j--;
        while (a[i] <= flag && i < j)
            i++;
        if (i < j)
        {
            temp = a[i];
            a[i] = a[j];
            a[j] = temp;
        }
    }
    a[left] = a[i];
    a[i]    = flag;
    quick(left, i-1);
    quick(i+1, right);
    return;
}

int main() {
    int i;
    printf("请输入排序个数\n");
    scanf("%d", &n);
    for (i = 1; i <= n; i++) {
        scanf("%d", &a[i]);
    }

    quick(1, n);

    for (i = 1; i <= n; i++) {
        printf("%d ",a[i]);
    }
    return 0;
}

File: Sort/quick.php
<?php
/**
 * 快速排序
 * @param array $data
 * @throws Exception
 */
function quick(array $data = array())
{
    $num = count($data);
    if ($num <= 1)
        return $data;
    $left = $right = [];
    for ($i = 1; $i < $num; $i++)
    {
        if ($data[$i] < $data[0])
            $left[] = $data[$i];
        else
            $right[] = $data[$i];
    }
    $left = quick($left);
    $right = quick($right);
    return array_merge($left, [$data[0]], $right);
}

include "../Config.php";
try{
    var_dump(quick($sort_data));
}catch (Exception $e){
    var_dump($e->getMessage());
}